ontolearner/evaluation/metrics.py

```diff
@@ -0,0 +1,240 @@
+
+    # F1 score: Harmonic mean of precision and recall
+    f1 = 2 * (precision * recall) / (precision + recall) if (precision + recall) > 0 else 0
+
+    # Exact match: True if predicted and ground truth sets are exactly the same
+    exact_match = 1.0 if predicted_set == ground_truth_set else 0.0
+
+    return {
+        'precision': precision,
+        'recall': recall,
+        'f1': f1,
+        'exact_match': exact_match
+    }
+
+
+def calculate_taxonomy_metrics(predicted: bool, ground_truth: bool) -> Dict[str, float]:
+    """
+    Calculate metrics for taxonomy relation prediction
+    """
+    # For taxonomy, it's a binary classification task
+    accuracy = 1.0 if predicted == ground_truth else 0.0
+
+    # True positive, false positive, etc.
+    tp = 1.0 if predicted and ground_truth else 0.0
+    fp = 1.0 if predicted and not ground_truth else 0.0
+    fn = 1.0 if not predicted and ground_truth else 0.0
+
+    # Calculate precision and recall for F1
+    precision = tp / (tp + fp) if (tp + fp) > 0 else 0
+    recall = tp / (tp + fn) if (tp + fn) > 0 else 0
+    f1 = 2 * (precision * recall) / (precision + recall) if (precision + recall) > 0 else 0
+
+    return {
+        'accuracy': accuracy,
+        'f1': f1
+    }
+
+
+def calculate_non_taxonomy_metrics(predicted: str, ground_truth: str) -> Dict[str, float]:
+    """
+    Calculate metrics for non-taxonomy relation prediction
+    """
+    # Exact match between relation types
+    exact_match = 1.0 if predicted.lower() == ground_truth.lower() else 0.0
+
+    # String similarity (optional if you want to give partial credit)
+    from difflib import SequenceMatcher
+    similarity = SequenceMatcher(None, predicted.lower(), ground_truth.lower()).ratio()
+
+    return {
+        'exact_match': exact_match,
+        'similarity_score': similarity
+    }
+
+
+def aggregate_metrics(results: List[Dict[str, Any]], task: str) -> Dict[str, float]:
+    """
+    Aggregate metrics across multiple examples
+    """
+    metrics = {}
+
+    if not results:
+        return metrics
+
+    if task == "term-typing":
+        metrics['avg_precision'] = np.mean([r['precision'] for r in results])
+        metrics['avg_recall'] = np.mean([r['recall'] for r in results])
+        metrics['avg_f1'] = np.mean([r['f1'] for r in results])
+        metrics['avg_exact_match'] = np.mean([r['exact_match'] for r in results])
+
+    elif task == "taxonomy-discovery":
+        metrics['avg_accuracy'] = np.mean([r['accuracy'] for r in results])
+        metrics['avg_f1'] = np.mean([r['f1'] for r in results])
+
+    elif task == "non-taxonomy-discovery":
+        metrics['avg_exact_match'] = np.mean([r['exact_match'] for r in results])
+        metrics['avg_similarity'] = np.mean([r['similarity_score'] for r in results])
+
+    metrics['num_samples'] = len(results)
+
+    return metrics
+
+
+def plot_model_comparison(summary_df, path: Path):
```

Comment:
Visualization should not appear here (inside the metrics), please make another script in the `evaluation` dir and give it an appropriate name!